src/zabapgit_repo_impl.prog.abap

```diff
@@ -663,4 +690,37 @@ CLASS lcl_repo_srv IMPLEMENTATION.
 
   ENDMETHOD. "is_repo_installed
 
+  METHOD switch_repo_type.
+
+    DATA lo_repo TYPE REF TO lcl_repo.
+
+    FIELD-SYMBOLS <repo> LIKE LINE OF mt_list.
+
+    lo_repo = get( iv_key ).
+    READ TABLE mt_list ASSIGNING <repo> FROM lo_repo.
+    ASSERT sy-subrc IS INITIAL.
+    ASSERT iv_offline <> lo_repo->ms_data-offline.
+
+    IF iv_offline = abap_true. " On-line -> OFFline
+      lo_repo->set(
+        iv_url         = lcl_url=>name( lo_repo->ms_data-url )
+        iv_branch_name = ''
+        iv_sha1        = ''
+        iv_head_branch = ''
+        iv_offline     = abap_true
+      ).
```

Comment:
we need the abaplint errors fixed, or the travis build will not build the abapmerge'd abap file